repository/managers.py

```diff
@@ -40,18 +40,70 @@ def collections(self, groupId):
             return response.json()  # Return the Collections data
         else:
             response.raise_for_status()
-    
+
     def collection_items(self, groupId, collectionId):
-        """Fetch collection items from the repository's endpoint"""
+        """
+        Fetch all items from a specific collection in a group.
+
+        This function retrieves all items from the specified collection in the repository.
+        It gathers all pages of items, combines them into a single JSON object, and includes the group details.
+
+        Args:
+            groupId: The ID of the group in the repository.
+            collectionId: The ID of the collection within the group.
+
+        Returns:
+            A dictionary containing:
+                - "group": Details about the group.
+                - "items": A list of all items in the specified collection.
+        """
         headers = auth.citesphere_auth(self.user, self.repository)
-        url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/collections/{collectionId}/items/"
-        response = requests.get(url, headers=headers)
         
-        if response.status_code == 200:
-            return response.json()
-        else:
+        base_url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/collections/{collectionId}/items/"
+        collections_url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/collections/"
+
+        # Fetch the collection details to determine the total number of items
+        collections_response = requests.get(collections_url, headers=headers)
+        if collections_response.status_code != 200:
+            collections_response.raise_for_status()
+
+        # Parse the response to find the specific collection and get the number of items in the collection
+        collections_data = collections_response.json().get('collections', [])
+        collection_num_items = 0
+        group_info = collections_response.json().get('group', {})
+        for collection in collections_data:
+            if collection.get('key') == collectionId:
+                collection_num_items = collection.get('numberOfItems', 0)
+                break
```

Comment:
alright, we need to change that. there needs to be a citesphere endpoint to get the information about a specific collection. let me see if I merge now or wait till that is there.